src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveWorkflow.cs

```diff
@@ -150,18 +168,45 @@ public sealed class RInteractiveWorkflow : IRInteractiveWorkflow {
         }
 
         public void Dispose() {
-            if (_disposed) {
-                return;
-            }
-            _disposed = true;
-
-            _debuggerModeTracker.EnterBreakMode -= DebuggerEnteredBreakMode;
-            _debuggerModeTracker.LeaveBreakMode -= DebuggerLeftBreakMode;
-            _activeTextViewTracker.LastActiveTextViewChanged -= LastActiveTextViewChanged;
-            RSession.Disconnected -= RSessionDisconnected;
-            Operations.Dispose();
-            Connections.Dispose();
+            _disposableBag.TryDispose();
             _onDispose();
         }
+
+        private void OnSettingsChanged(object sender, PropertyChangedEventArgs e) {
+            if (e.PropertyName == nameof(IRSettings.CranMirror)) {
+                SetMirrorToSession().DoNotWait();
+            } else if (e.PropertyName == nameof(IRSettings.RCodePage)) {
+                SetSessionCodePage().DoNotWait();
+            } 
+        }
+
+        private async Task SetMirrorToSession() {
+            string mirrorName = _settings.CranMirror;
+            string mirrorUrl = CranMirrorList.UrlFromName(mirrorName);
+
+            foreach (var s in RunningSessions) {
+                try {
+                    using (var eval = await s.BeginEvaluationAsync()) {
+                        await eval.SetVsCranSelectionAsync(mirrorUrl);
```

Comment:
>SetVsCranSelectionAsync [](start = 35, length = 23)

I think you can call `SetVsCranSelectionAsync` on a session itself, without `BeginEvaluationAsync`